ea-factories/src/factories/balance.ts

```diff
@@ -17,14 +11,20 @@ const ERROR_MISSING_ADDRESS = 'No Operation: address param is missing'
 
 export type IsSupported = (coin: string, chain: string) => boolean
 export type BalancesResponse = DataResponse<Account[], any>
-export type GetBalance = (account: Account, config: BalanceConfig) => Promise<BalancesResponse>
-export type GetBalances = (accounts: Account[], config: BalanceConfig) => Promise<BalancesResponse>
-
-export type BalanceConfig = Config & {
+export type GetBalance<C extends Config = Config> = (
+  account: Account,
+  config: BalanceConfig<C>,
+) => Promise<BalancesResponse>
+export type GetBalances<C extends Config = Config> = (
+  accounts: Account[],
+  config: BalanceConfig<C>,
+) => Promise<BalancesResponse>
+
+export type BalanceConfig<C extends Config = Config> = C & {
   confirmations?: number
   isSupported: IsSupported
-  getBalance?: GetBalance
-  getBalances?: GetBalances
+  getBalance?: GetBalance<C>
+  getBalances?: GetBalances<C>
 }
```

Comment:
I assume you noticed there is something wrong here 😅

This exploded in complexity, unnecessarily IMHO.

A hint you can use is that the API should probably be amended to look more like this:

```javascript
export type GetBalances = (accounts: Account[], confirmations?: number) => Promise<BalancesResponse>
```

We don't configure the function calls, we configure the adapter. Also not sure we even need the single returning `GetBalance`, as we can support it all by using `GetBalances`.

Reply:
This happened because the Config needed to be extended to include new per provider env variables such as `API_SECRET`. This same config gets used in the factory, so I made everything a generic to take in possibly extended Conifgs, but also fall back to the default Config if not specified.

Not sure that I understand how we could make the `GetBalances` balance request without the `Config`, but I'll put some thought into how I could refactor & clean this up.